### PR TITLE
Fix accept/reject remote share action #31301

### PR DESCRIPTION
**Pull request by MSe1969**
This is the fix for the described problem in 
https://github.com/nextcloud/server/issues/31301

### CODE REVIEW

**Comment by szaimen** [comment]
@MSe1969 thanks for fixing this!

Can you please solve DCO? Thank you! :)

**Comment by MSe1969** [comment]
> @MSe1969 thanks for fixing this!
> 
> Can you please solve DCO? Thank you! :)
Done.

**Comment by szaimen** [comment]
Thank you! :)

File: apps/files_sharing/js/app.js [hunk 1]
```diff
@@ -303,7 +303,7 @@ OCA.Sharing.App = {
 				if (context.$file.attr('data-remote-id')) {
 					shareBase = 'remote_shares/pending'
 				}
-				$.post(OC.linkToOCS('apps/files_sharing/api/v1/shares/pending', 2) + shareId)
+				$.post(OC.linkToOCS('apps/files_sharing/api/v1/remote_shares/pending', 2) + shareId)
```
Comment (szaimen) [inline apps/files_sharing/js/app.js:306]: the question for me is if internal shares will still work with this fix or if we need to add a switch or condition based on internal or federated sharing...

File: apps/files_sharing/js/app.js [hunk 2]
```diff
@@ -334,7 +334,7 @@ OCA.Sharing.App = {
 				}
 
 				$.ajax({
-					url: OC.linkToOCS('apps/files_sharing/api/v1/shares', 2) + shareId,
+					url: OC.linkToOCS('apps/files_sharing/api/v1/remote_shares', 2) + shareId,
```
Comment (szaimen) [inline apps/files_sharing/js/app.js:337]: same as above

File: apps/files_sharing/js/app.js (hunk 1 above)
Comment (nickvergessen) [reply apps/files_sharing/js/app.js:306]: Test it?

Comment (szaimen) [reply apps/files_sharing/js/app.js:306]: maybe on monday ;)

Comment (MSe1969) [reply apps/files_sharing/js/app.js:306]: Hmmm... which internal shares trigger an accept/reject action? As far as I have experienced it, only external shares ask for confirmation, or did I overlook any functionality? Depending on the answer to that topic, there is either the variable shareBase to be used again, or some dead code to be cleaned up...

File: apps/files_sharing/js/app.js (hunk 2 above)
Comment (MSe1969) [reply apps/files_sharing/js/app.js:337]: likewise

File: apps/files_sharing/js/app.js (hunk 1 above)
Comment (szaimen) [reply apps/files_sharing/js/app.js:306]: After having a deeper look at the logic I think this should work:
```suggestion
				$.post(OC.linkToOCS('apps/files_sharing/api/v1/' + shareBase, 2) + shareId)
```
Please report back if it works! Thank you! :)

File: apps/files_sharing/js/app.js (hunk 2 above)
Comment (szaimen) [reply apps/files_sharing/js/app.js:337]: Same as above:
```suggestion
				$.post(OC.linkToOCS('apps/files_sharing/api/v1/' + shareBase, 2) + shareId)
```

Comment (szaimen) [reply apps/files_sharing/js/app.js:337]: same as above:
```suggestion
					url: OC.linkToOCS('apps/files_sharing/api/v1/' + shareBase, 2) + shareId,
```

Comment (szaimen) [reply apps/files_sharing/js/app.js:337]: I guess three lines above we need to adjust `shareBase = 'remote_shares/pending'` to `shareBase = 'remote_shares'`

Comment (szaimen) [reply apps/files_sharing/js/app.js:337]: Please report back if that works! Thank you! :)

File: apps/files_sharing/js/app.js (hunk 1 above)
Comment (nickvergessen) [reply apps/files_sharing/js/app.js:306]: > which internal shares trigger an accept/reject action?

I guess you are looking for this:
https://github.com/nextcloud/server/blob/b8b4d247b4382d7c51323976a3a17e1416dcfe0a/config/config.sample.php#L1562-L1571

Comment (MSe1969) [reply apps/files_sharing/js/app.js:306]: OK, clear and understood (did not know about that)

Comment (MSe1969) [reply apps/files_sharing/js/app.js:306]: Quick question, before messing things up: 
Should I initiate another force-push to the PR or can I accept the proposal in here ("commit suggestion")?

Comment (szaimen) [reply apps/files_sharing/js/app.js:306]: Honestly both is fine but please don't forget the sign-off :)

Force-pushing is probably a bit nicer though :)

Comment (MSe1969) [reply apps/files_sharing/js/app.js]: OK, force-pushed correction - my own tests for remote share and local share (I temporarily have set the sharing.force_share_accept option) now also works. So good to go from my side, but please also test on your end.

**Comment by szaimen** [comment]
Looks good to me :)
@MSe1969 can you please rebase, compile the js and push it to this branch? Then CI should be green :)

**Review by szaimen — APPROVED** [review]
Thanks! Looks good to me but didn't test

**Comment by MSe1969** [comment]
Question: 
As this would also apply to the current stable version 23 - will you centrally take care about that or do I need to also make the same PR against the stable23 branch?

**Review by Pytal — APPROVED** [review]

**Comment by szaimen** [comment]
Failure is unrelated

**Comment by szaimen** [comment]
https://github.com/nextcloud/server/pull/31949

**Pull request merged**